order/src/main/java/com/yas/order/service/ProductService.java

```diff
@@ -1,80 +1,120 @@
 package com.yas.order.service;
 
+import com.yas.commonlibrary.exception.NotFoundException;
 import com.yas.order.config.ServiceUrlConfig;
+import com.yas.order.utils.AuthenticationUtils;
 import com.yas.order.viewmodel.order.OrderItemVm;
 import com.yas.order.viewmodel.order.OrderVm;
+import com.yas.order.viewmodel.product.ProductCheckoutListVm;
+import com.yas.order.viewmodel.product.ProductGetCheckoutListVm;
 import com.yas.order.viewmodel.product.ProductQuantityItem;
 import com.yas.order.viewmodel.product.ProductVariationVm;
 import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
 import io.github.resilience4j.retry.annotation.Retry;
 import java.net.URI;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import lombok.RequiredArgsConstructor;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.security.oauth2.jwt.Jwt;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestClient;
 import org.springframework.web.util.UriComponentsBuilder;
 
 @Service
 @RequiredArgsConstructor
 public class ProductService extends AbstractCircuitBreakFallbackHandler {
+
     private final RestClient restClient;
     private final ServiceUrlConfig serviceUrlConfig;
 
     @Retry(name = "restApi")
     @CircuitBreaker(name = "restCircuitBreaker", fallbackMethod = "handleProductVariationListFallback")
     public List<ProductVariationVm> getProductVariations(Long productId) {
-        final String jwt = ((Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal())
-            .getTokenValue();
+        final String jwt = AuthenticationUtils.getCurrentJwtTokenValue();
+
         final URI url = UriComponentsBuilder
-            .fromHttpUrl(serviceUrlConfig.product())
-            .path("/backoffice/product-variations/" + productId)
-            .buildAndExpand()
-            .toUri();
+                .fromHttpUrl(serviceUrlConfig.product())
+                .path("/backoffice/product-variations/" + productId)
+                .buildAndExpand()
```

Comment:
Please use this to append path variables instead of `.path("/backoffice/product-variations/" + productId)`
```java
final URI url = UriComponentsBuilder
    .fromHttpUrl(serviceUrlConfig.product())
    .path("/backoffice/product-variations/${productId}")
    .buildAndExpand(productId)
    .toUri();
```